### PR TITLE
[CXP-863] Sync granular RBAC permissions

### PR DESCRIPTION
**Pull request by agustin-conductor**
Closes [CXP-863](https://linear.app/ductone/issue/CXP-863/baton-kubernetes-sync-granular-rbac-permissions-opt-in). Also addresses the CXP-376 QA finding (verb entitlements that never receive grants) and the customer request behind CXH-1976 / CE-62.

Membership already answered *who holds a role*. What a role permits was profile text only. This makes it an edge, so "who can create pods in team-a" is a query rather than a reading exercise:

```
kube_user / kube_group / service_account
        │  holds the role, in a scope     (role / cluster_role / role_assignment)
        ▼
cluster_role: team-a-editor
        │  grant: create                  ← what this PR adds
        ▼
api_resource: core:pods@team-a            "pods in team-a"
```

## The model

**`api_resource`** — one resource per authorization target an RBAC rule names, keyed on the tuple the API server's own authorizer evaluates: `(apiGroup, resource[/subresource], scope)`.

```
core:pods@team-a                     pods in team-a          (child of the namespace)
core:pods@*                          pods cluster-wide       (child of the cluster singleton)
core:pods/exec@team-a                a distinct API resource, not a verb on pods
core:secrets:app-db-password@team-a  narrowed by a resourceNames rule
*:*@*                                what cluster-admin's wildcard rule grants
_url:/healthz@*                      a non-resource URL
```

This replaces the nine per-type wildcards (`pod:*`, `secret:*`, …), which conflated a class of objects with a scope, were the only unparented resources in the connector, and were minted whether or not any rule referenced them. The scope now comes from the binding, never the role, so the same ClusterRole produces `core:pods@*` through a ClusterRoleBinding and `core:pods@team-a` through a RoleBinding.

The API group is part of the ID because a resource plural is not unique: a stock cluster serves `events` in both the core and `events.k8s.io` groups, ClusterRole `edit` names `deployments` in both `apps` and the long-dead `extensions`, and CRDs collide freely (`gateways` exists in both `networking.istio.io` and `gateway.networking.k8s.io`). Display names hide it for core kinds.

**Object types** (pod, secret, configmap, deployment, statefulset, daemonset, node, namespace, service_account) declare a fixed permission set per type through `StaticEntitlements`. An entitlement is a capability and a grant is who holds it, so a pod declares `delete` whether or not anything confers it — and nothing in C1 referencing that entitlement churns when RBAC changes. Their grants carry the object-addressable half of every rule reaching their type in their scope, plus any rule naming them explicitly.

**Grant principals** are role-shaped, never identities: Kubernetes stores the permission on the role. Each grant is `GrantExpandable` through the principal's own membership entitlement, so the SDK's expansion phase carries it to the subjects. The principal has to be whichever resource carries a scope-matching membership entitlement, because expansion requires the expandable entitlement to live on the grant's own principal:

| mode | binding | principal | expands through |
| --- | --- | --- | --- |
| flat | ClusterRoleBinding | `cluster_role` | `all:member` |
| flat | RoleBinding in team-a | `cluster_role` | `team-a:member` |
| `--use-role-assignments` | either | `role_assignment` | `assigned` |
| any | namespaced Role | `role` | `member` |

## Fidelity rules

Each of these was a bug caught against a live cluster, not a hypothetical:

- **Collection verbs stay on `api_resource`.** `list`, `watch` and `deletecollection` address a collection, not any member of it. `create` is object-addressable only for subresources — `kubectl exec mypod` is a `create` on `pods/exec` whose path carries the pod name.
- **Subresources keep their name** (`create:exec`, `get:log`, `update:scale`). The verb alone would read as permission to create the pod.
- **`resourceNames` rules get a narrowed target** rather than widening the class. Folding them in claimed that a role able to read one Secret could read every Secret in the namespace. Only verbs a named request can carry come along — `resourceNames` cannot gate `list`, which is the classic RBAC trap.
- **Rules Kubernetes ignores produce nothing.** A namespaced binding can never reach a cluster-scoped kind, and a wildcard over `*/scale` reaches Deployments but not Secrets (there is no `secrets/scale`). Both read the discovery endpoints, which need no extra access: `system:discovery` is bound to `system:authenticated` on every cluster.
- **Wildcards stay literal**, so `cluster-admin` costs one target and one entitlement instead of being expanded across the API surface.
- **`system:` cluster roles are excluded from the object layer**, where they were 78% of grants and reach every object of every type. They remain on the `api_resource` targets, where the cost is one edge per API resource rather than one per object. `--include-system-object-permissions` restores them.

## Removed

`generateWildcardResource` and the nine `*` resources; the fixed seven-verb sets on the workload builders; deployment `scale`/`rollback`, pod `exec`/`portforward` and service_account `impersonate`. Those were 300+ entitlements that could never be granted, because nothing in the cluster conferred them — the CXP-376 QA finding. `pods/exec` is now modelled as what it actually is.

## Breaking changes

Entitlement and resource IDs on the object types change, with no flag to opt out:

| before | after |
| --- | --- |
| `pod:<ns>/<name>:exec` | `pod:<ns>/<name>:create:exec` |
| `pod:<ns>/<name>:list` / `:watch` / `:create` | gone — collection verbs live on `api_resource` |
| `deployment:<ns>/<name>:scale` / `:rollback` | `deployment:<ns>/<name>:update:scale` (rollback has no `apps/v1` endpoint) |
| `service_account:<ns>/<name>:impersonate` | unchanged in name, now only granted where a rule confers it |
| `pod:<ns>/<name>:portforward` | `pod:<ns>/<name>:create:portforward` and `:get:portforward` |
| `statefulset:<ns>/<name>:scale` | `statefulset:<ns>/<name>:update:scale` |
| `pod:*`, `secret:*`, … (nine wildcards) | `api_resource:core:pods@*`, `core:secrets@*`, … |

The `cluster` singleton also loses its `enabled` gate, so it now syncs whenever the type is
selected rather than only under `--use-role-assignments`. It is one resource, and both the
sparse assignments and the cluster-wide `api_resource` targets anchor to it.

No access data is lost — every removed entitlement provably carried zero grants, which is the
CXP-376 QA finding — but anything in C1 referencing those IDs (a campaign, a request, a policy)
orphans on deploy.

## Configuration

No flag gates the permissions: selecting a resource type is what asks for its access data. One new flag, and it only narrows:

| flag | default | effect |
| --- | --- | --- |
| `--include-system-object-permissions` | `false` | also report what `system:` cluster roles permit on individual objects |

`api_resource` and `cluster` carry `OptInRequired` but are in the connector's default selection. **If you narrow `--sync-resource-types`, keep both** — `api_resource` is the only resource that can express a permission over a class of objects, and the collection verbs exist nowhere else, so dropping it makes "who can create pods in team-a" unanswerable.

Keep the principal types too (`role`, `cluster_role`, and `role_assignment` under the sparse model). Every permission grant's principal is one of those, so dropping one means the SDK filters those grants out (`pkg/sync/ingest_filter.go` drops a grant whose principal type is not scheduled) and the permission targets sync with entitlements and no grants — the silently-empty surface this PR removes. Documented in the README rather than validated in code: `--sync-resource-types` is an SDK runner flag, and `NewFromConfig` deliberately registers every type because the real selection arrives per sync task.

Pods carry `SkipSyncAnomalyDetection`: a rollout legitimately replaces every pod, and that drop is churn rather than an access regression.

## Measured on the kind fixture (8 namespaces, 74 cluster roles, 14 roles)

| | resources | entitlements | grants |
| --- | --- | --- | --- |
| before | 160 | 709 | 53 |
| default sync | 370 | 1,690 | 3,024 |
| every type selected, sparse model | 463 | 1,795 | 3,495 |

**Scale to weigh before large clusters.** The class layer grows with the API surface plus the breadth of the roles bound in each namespace — 129 cluster-wide targets and 59 in the one namespace with broad bindings here; a cluster with 50 namespaces each binding `view` and `edit` lands nearer 3,500 targets and ~20,000 entitlements. The object layer grows with objects, and pods are the type to watch at 25 entitlements each. Narrowing `--sync-resource-types` is the lever.

End to end, on the fixture:

```
G api_resource:core:pods@team-a:create        <- role:team-a/team-a-editor      # direct
                                              <- kube_user:editor@example.com   # expanded
                                              <- kube_group:platform-devs       # expanded
G pod:team-a/api-server:create:exec           <- role_assignment:ns:team-a:pod-debugger
G service_account:team-a/deployer:impersonate <- role_assignment:ns:team-a:sa-impersonator
G configmap:kube-public/cluster-info:update   <- role:kube-public/system:controller:bootstrap-signer
```

## Testing

`go build ./...`, `go test ./...` and `go vet ./pkg/... ./cmd/...` pass. New tests cover the class and object layers, both membership models, the narrowed `resourceNames` targets, the inert-rule and wildcard-subresource guards, the control-plane filter, paging, and the fixed-vs-derived entitlement contract.

Verified against a kind cluster with fixtures exercising the full verb surface (`test-cluster/rbac-fixtures.yaml` in the working notes): the classic set on namespaced types, `deployments/scale`, `pods/exec`/`portforward`/`log`, `serviceaccounts` impersonate + token, and the classic set on a cluster-scoped kind bound cluster-wide.

One unrelated environment note: `golangci-lint run ./...` currently panics inside its own `nilness` analyzer while processing a vendored package (`honnef.co/go/tools` v0.8.0-rc.1 against the local Go toolchain). `golangci-lint run --disable=staticcheck ./pkg/... ./cmd/...` reports 0 issues.

## Follow-ups, not in this PR

- `canModify` → `roleCanModify` rename and confirming the profile fields are filterable in C1.
- Porting to baton-eks / baton-aks / baton-gke, which override the role builders.
- `TypeScopedGrantsSyncer` for the object layer if the per-object `Grants()` round trips become the bottleneck on a large cluster (414 calls on the fixture; ~5,000 at 5,000 pods). It changes no data.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/connector/object_permissions.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+var objectPermissionSets = map[string][]string{
+	ResourceTypePod.Id: objectSlugs(
+		newSubresource("", verbGet, verbUpdate, verbPatch, verbDelete),
+		newSubresource("attach", verbCreate, verbGet),
+		newSubresource("binding", verbCreate),
+		newSubresource("ephemeralcontainers", verbGet, verbPatch, verbUpdate),
+		newSubresource("eviction", verbCreate),
+		newSubresource("exec", verbCreate, verbGet),
+		newSubresource("log", verbGet),
+		newSubresource("portforward", verbCreate, verbGet),
```
Comment (FeliLucero1) [inline pkg/connector/object_permissions.go:40]: **[Breaking]** heads up, this quietly renames/removes a bunch of entitlements that already shipped in v0.2.0 (which is live on the `latest` channel btw, not unreleased like the PR assumes) — pod `exec`→`create:exec`, `portforward`→`create:portforward`/`get:portforward`, bare `list`/`watch`/`create` gone everywhere, deployment/statefulset `scale`→`update:scale`, `rollback` dropped, and all 9 wildcard resources deleted. `portforward` and statefulset's `scale` rename aren't even in the PR's own breaking-changes table. No alias path used either (SDK has `WithAliases` for exactly this, unused here) — any C1 campaign/request referencing the old IDs just orphans on deploy.

File: pkg/connector/resource_types.go [hunk 2]
```diff
@@ -104,6 +165,7 @@ func DefaultSyncResourceTypeIDs() []string {
 		ResourceTypeKubeGroup.Id,
 		ResourceTypeCluster.Id,
 		ResourceTypeRoleAssignment.Id,
+		ResourceTypeAPIResource.Id,
```
Comment (FeliLucero1) [inline pkg/connector/resource_types.go]: **[Process]** this forces `api_resource` into the default sync even though it's marked `OptInRequired` — turns out that annotation only affects the platform UI, not this connector's own default list, so a zero-flag upgrade already gets 2.4x entitlements / 57x grants per your own fixture numbers. Our repo's own `review-breaking-changes.md` says changes like this (entitlement slug/resource ID changes) need a flag + lead approval before merge — think this needs one of those before it ships.

File: pkg/connector/api_resource.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+// permissionClassCursor is the page token: the last class emitted, rather than
+// an index into the derived list.
+//
+// An index does not survive the cross-process resume the rest of this connector
+// is built for. A resumed sync rebuilds the class list from current cluster
+// state, so if any class sorting before the index disappeared meanwhile,
+// everything after it shifts down and whichever class lands on the old index is
+// never emitted — silently. Resuming from the first class after a recorded key
+// is self-correcting.
+type permissionClassCursor struct {
```
Comment (FeliLucero1) [inline pkg/connector/api_resource.go:212]: **[Nit]** this cursor/pagination logic is basically copy-pasted from `role_assignment.go`'s `assignmentCursor` (even the doc comments match almost word for word) — worth pulling into one shared generic helper instead of maintaining two copies of the same resume logic.

File: pkg/connector/permission_index.go [hunk 4]
```diff
@@ -0,0 +1,1161 @@
...
+}
+
+// objectPermissions returns everything one object carries: the permissions every
+// object of its type carries in its scope, plus any a resourceNames rule confers
+// on it specifically.
+//
+// Cluster-wide rules are merged in for a namespaced object, because a
+// ClusterRoleBinding reaches every namespace — including the one this object
+// lives in.
+func (r *permissionResolver) objectPermissions(
```
Comment (FeliLucero1) [inline pkg/connector/permission_index.go:441]: **[Perf]** this redoes the merge/dedupe/sort from scratch on every single object, but the cluster+namespace-scope part is identical for every sibling object in the same (type, namespace) — only `resourceNames` really varies per object. Worth caching per-scope so this doesn't multiply by object count at scale (you already called out the ~5k pods case in the follow-ups, this would help with that).

File: pkg/connector/cluster.go [hunk 5]
```diff
@@ -27,25 +27,18 @@ type clusterBuilder struct {
...
 }
 
-// List returns the single cluster resource, or nothing when the sparse model is
-// off.
+// List returns the single cluster resource.
+//
+// Emitted whenever the type is selected, which is the only gate it needs: it is
+// one resource, and both the sparse role assignments and the cluster-wide
+// api_resource targets point at it as their scope or parent.
 func (c *clusterBuilder) List(_ context.Context, _ *v2.ResourceId, _ rs.SyncOpAttrs) ([]*v2.Resource, *rs.SyncOpResults, error) {
```
Comment (FeliLucero1) [inline pkg/connector/cluster.go:41]: **[FYI]** the old `enabled` gate is gone so Cluster now always shows up once the type is selected — makes sense given the new api_resource parenting, just didn't see it called out in the breaking-changes list, might be worth a one-liner there.

File: pkg/connector/pod.go [hunk 6]
```diff
@@ -119,61 +111,35 @@ func podResource(pod *corev1.Pod) (*v2.Resource, error) {
...
+// entitlements through StaticEntitlements instead.
+func (p *podBuilder) Entitlements(_ context.Context, _ *v2.Resource, _ rs.SyncOpAttrs) ([]*v2.Entitlement, *rs.SyncOpResults, error) {
 	return nil, nil, nil
 }
 
+// Grants returns the roles conferring those named-object permissions, expandable
+// through each one's membership entitlement so the subjects holding the role
+// inherit the permission.
+func (p *podBuilder) Grants(ctx context.Context, resource *v2.Resource, opts rs.SyncOpAttrs) ([]*v2.Grant, *rs.SyncOpResults, error) {
+	grants, err := objectGrants(ctx, p.perms, opts, ResourceTypePod, resource)
```
Comment (FeliLucero1) [inline pkg/connector/pod.go:132]: **[Nit]** small one — `if err != nil { return nil, nil, err }; return grants, nil, nil` can just be `return grants, nil, err` since `objectGrants` already returns nil grants on error. Same pattern repeats in secret/configmap/deployment/daemonset/statefulset/node/namespace/serviceaccount too.

File: pkg/connector/permission_index.go (hunk 4 above)
Comment (agustin-conductor) [reply pkg/connector/permission_index.go]: Addressed in 4abc71aa — the cluster+namespace union is precomputed per (type, scope) at index build time, so an object no rule names gets the shared map back with zero allocation per object.

File: pkg/connector/api_resource.go (hunk 3 above)
Comment (agustin-conductor) [reply pkg/connector/api_resource.go]: Addressed in 4abc71aa — one generic `pageBounds`, rationale stated once. Also added a multi-page test for role assignments, which had none.

File: pkg/connector/cluster.go (hunk 5 above)
Comment (agustin-conductor) [reply pkg/connector/cluster.go]: Good catch — added to the breaking-changes list in the PR body.

File: pkg/connector/pod.go (hunk 6 above)
Comment (agustin-conductor) [reply pkg/connector/pod.go]: Addressed in 4abc71aa, all nine builders.

File: pkg/connector/permission_index.go [hunk 7]
```diff
@@ -0,0 +1,1161 @@
...
+
+// addNamedObjects records the edges a resourceNames rule puts on the real
+// objects it names.
+func (b *permissionIndexBuilder) addNamedObjects(
+	rule rbacv1.PolicyRule,
+	resource, kind string,
+	verbs []string,
+	src ruleSource,
+) {
+	resourceType, ok := instanceResourceTypes[kind]
```
[the remark this answers is not in the document]
Comment (agustin-conductor) [reply pkg/connector/permission_index.go]: Valid — fixed in 4abc71aa. Matches through `instanceKinds` now, same as the class layer, and `instanceResourceTypes` is deleted since it was the duplicate table that allowed the drift.

File: pkg/connector/permission_index.go [hunk 8]
```diff
@@ -0,0 +1,1161 @@
...
+		for _, verb := range verbs {
+			if !addressesObject(verb, resource) {
+				continue
+			}
+			bySlug, ok := b.objects[scope]
+			if !ok {
+				bySlug = map[string]map[principalRef]struct{}{}
+				b.objects[scope] = bySlug
+			}
+			slug := objectEntitlementSlug(resource, verb)
```
[the remark this answers is not in the document]
Comment (agustin-conductor) [reply pkg/connector/permission_index.go]: Fixed in 4abc71aa — `*` on a subresource expands to the verbs that endpoint accepts (`create:exec`, `get:exec`) rather than an undeclared `*:exec`.

File: pkg/connector/permission_index.go [hunk 9]
```diff
@@ -0,0 +1,1156 @@
...
+func discoverAPIResources(ctx context.Context, client kubernetes.Interface) (map[schema.GroupResource]apiResourceInfo, error) {
+	l := ctxzap.Extract(ctx)
+
+	_, lists, err := client.Discovery().ServerGroupsAndResources()
+	if err != nil {
+		if !discovery.IsGroupDiscoveryFailedError(err) {
+			return nil, fmt.Errorf("failed to discover API resources: %w", err)
+		}
+		l.Debug("partial API discovery failure; undiscovered kinds are treated as unknown", zap.Error(err))
+	}
```
[the remark this answers is not in the document]
Comment (agustin-conductor) [reply pkg/connector/permission_index.go]: Valid — fixed in 0d410b11. `discoverAPIResources` no longer returns an error at all, so it cannot fail a sync; the regression test asserts the induced error is *not* `ErrGroupDiscoveryFailed`, which is the distinction the bug turned on.

File: pkg/connector/permission_index.go [hunk 10]
```diff
@@ -0,0 +1,1161 @@
...
+// satisfy is reported instead of dropped — the documented trade. hasSubresource
+// reports every subresource as present, so a wildcard rule over "*/scale" spreads
+// to kinds without a scale endpoint, and those slugs are then dropped by the
+// declared set in object_permissions.go, which no cluster state can widen.
+func discoverAPIResources(ctx context.Context, client kubernetes.Interface) map[schema.GroupResource]apiResourceInfo {
+	l := ctxzap.Extract(ctx)
+
+	_, lists, err := client.Discovery().ServerGroupsAndResources()
+	if err != nil {
+		l.Debug("API discovery incomplete; undiscovered kinds are treated as unknown",
```
[the remark this answers is not in the document]
Comment (agustin-conductor) [reply pkg/connector/permission_index.go]: Keeping Debug: there are no `Warn` calls anywhere in `pkg/connector` by convention. Happy to move it to `Info` if visibility at default level is preferred.

File: pkg/connector/resource_types.go [hunk 11]
```diff
@@ -104,6 +165,7 @@ func DefaultSyncResourceTypeIDs() []string {
 		ResourceTypeKubeGroup.Id,
```
[the remark this answers is not in the document]
Comment (agustin-conductor) [reply pkg/connector/resource_types.go]: Documented rather than validated — grants with an unscheduled principal type are dropped by `ingest_filter.go`, so the failure mode is a silently empty permission surface rather than dangling references. README now says to keep `role`/`cluster_role`/`role_assignment` selected. `NewFromConfig` cannot validate it: `--sync-resource-types` is an SDK runner flag and the real selection arrives per sync task.

File: README.md [hunk 12]
```diff
@@ -85,6 +86,199 @@ baton-kubernetes --use-role-assignments \
...
+`api_resource` grows with the cluster's API surface and the breadth of the roles bound in
+each namespace, not with object counts. On the 8-namespace, 71-ClusterRole test cluster:
+198 resources, 577 entitlements, 1,172 role-held grants and 1,310 expanded identity grants, of
+which 12 resources are `resourceNames`-narrowed targets.
+
+Object-level permissions scale with object count instead, and because the entitlement set is
+fixed per type, every object of a synced type carries its full set whether or not anything
+grants it. A default sync — the class layer plus namespaces and service accounts, no workload
+types — is 370 resources, 1,690 entitlements and 3,024 grants on the fixture; selecting every
+type takes it to 463, 1,795 and 3,495.
```
[the remark this answers is not in the document]
Comment (agustin-conductor) [reply README.md]: Fixed in 0d410b11 — 1,795 was a typo for 1,513, cluster-role count corrected. Also removed the fixture-specific figures from the README entirely since it is user-facing; they live in `docs-info.md` now.

File: pkg/connector/permission_index.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (agustin-conductor) [reply pkg/connector/permission_index.go]: Follow-up: this fix caused a regression the next review caught — see the reply on :902.

File: pkg/connector/permission_index.go [hunk 13]
```diff
@@ -0,0 +1,1244 @@
...
+
+	for _, kind := range instanceKinds {
+		if !kind.matches(group, resource) {
+			continue
+		}
+		if !b.hasSubresource(kind, subresource) {
+			// Same reasoning as the class layer: a rule over a subresource this
+			// kind does not have authorizes nothing against it.
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (agustin-conductor) [reply pkg/connector/permission_index.go]: Confirmed and fixed in 9d119fcf. Put the guard in `instanceObjectID` rather than the loop, since it already refuses the mirror case (a namespaced kind named cluster-wide), so the invariant stays in one place.

File: pkg/connector/role_assignment_test.go [hunk 14]
```diff
@@ -372,3 +372,41 @@ func TestRoleAssignmentStaticEntitlement(t *testing.T) {
...
+		if results == nil || results.NextPageToken == "" {
+			break
+		}
+		token = results.NextPageToken
+	}
+
+	sort.Strings(ids)
+	assert.Equal(t, []string{"cluster:edit", "cluster:view", "ns:team-a:admin", "ns:team-b:view"}, ids,
+		"every pair exactly once, in order, across single-item pages")
+}
```
[the remark this answers is not in the document]
Comment (agustin-conductor) [reply pkg/connector/role_assignment_test.go]: Fixed in 9d119fcf — sort removed, compares in key order now.

File: pkg/connector/resource_types.go (hunk 2 above)
Comment (mateoHernandez123) [inline pkg/connector/resource_types.go]: api_resource is marked OptInRequired, and optInAnnotations() says that keeps it out of the default sync (CLI needs an explicit --sync-resource-types). This list still includes it, so a no-flag CLI run emits the new permission graph. CXP-863 is default-off with unchanged output when the gate is off; type selection is the right gate, but only if the type stays out of the default set.

Drop ResourceTypeAPIResource.Id from DefaultSyncResourceTypeIDs (keep the type registered and OptInRequired). Pattern: https://github.com/ConductorOne/baton-kubernetes/blob/9d119fcfd2948bf71c138ac4de619a89e02a12d9/pkg/connector/resource_types.go#L50-L54

File: pkg/connector/resource_types.go [hunk 15]
```diff
@@ -53,6 +76,25 @@ var (
...
+	// namespace) — which is also why it can carry pods/exec, an API resource
+	// with no objects at all.
+	//
+	// It carries no trait, deliberately, for the same reason as the cluster
+	// singleton: a role trait here would count a permission as classic access
+	// in C1's access-model derivation.
+	ResourceTypeAPIResource = &v2.ResourceType{
+		Id:          "api_resource",
+		DisplayName: "API Resource",
+		Annotations: optInAnnotations(),
```
Comment (mateoHernandez123) [inline pkg/connector/resource_types.go]: Suggestion / question: this type is OptInRequired, so C1 keeps it off until the UI selection opts in. DefaultSyncResourceTypeIDs still includes it, which is only the no-flag CLI default — a bare CLI run emits the new permission graph, while a C1 tenant does not.

Was that split intentional for local/on-prem CLI, or would you rather drop api_resource from DefaultSyncResourceTypeIDs so CLI matches C1 (keep the type registered + OptInRequired)? Pattern: https://github.com/ConductorOne/baton-kubernetes/blob/9d119fcfd2948bf71c138ac4de619a89e02a12d9/pkg/connector/resource_types.go#L50-L54

**Review by luisina-santos — APPROVED** [review]

**Review by mateoHernandez123 — APPROVED** [review]
